src/software/thunderscope/common/proto_configuration_widget.py

```diff
@@ -58,9 +70,187 @@ def __init__(
         self.param_group.sigTreeStateChanged.connect(self.__handle_parameter_changed)
         self.param_tree.setAlternatingRowColors(False)
 
+        self.save_hbox_top, self.save_hbox_bottom, self.line_box = self.create_widget()
+
+        layout.addLayout(self.save_hbox_top)
+        layout.addLayout(self.save_hbox_bottom)
+
+        layout.addWidget(QLabel("Edit TBotsConfig"))
         layout.addWidget(self.search_query)
         layout.addWidget(self.param_tree)
 
+        self.run_onetime_async(3, self.send_proto_to_fullsystem)
+
+    def run_onetime_async(self, time_in_seconds: float, func: Callable):
+        """
+        starting a timer that runs a given function after a given
+        amount of seconds one time asynchronously.
+
+        :time_in_seconds: the amount of time in seconds
+        :func: the function that is going to be ran
+        """
+        self.timer = QTimer()
+        self.timer.setSingleShot(True)
+        self.timer.timeout.connect(func)
+        self.timer.start(time_in_seconds * MILLISECONDS_PER_SECOND)
+
+    def create_widget(self):
+        """
+        Creating widgets that are used to load, save parameters
+
+        :return: the top of the layout, the bottom of the layout, and edit box widget
+        """
+        save_button = QPushButton("Save Parameters")
+        load_proto_button = QPushButton("Load Parameters")
```

Comment:
ideally save and load would pop up a dialog box

Reply:
done.